src/FeeRouter/RewardReceiver.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.16;
+
+interface IArbSys {
+    function withdrawEth(
+        address destination
+    ) external payable returns (uint256);
+}
+
+contract RewardReceiver {
+    address immutable parentChainTarget;
+    uint256 immutable minDistributionIntervalSeconds;
+    uint256 public nextDistribution;
+
+    constructor(
+        address _parentChainTarget,
+        uint256 _minDistributionIntervalSeconds
+    ) {
+        parentChainTarget = _parentChainTarget;
+        minDistributionIntervalSeconds = _minDistributionIntervalSeconds;
+    }
+
+    receive() external payable {
+        sendFunds();
+    }
+
+    function sendFunds() public {
+        // if distributing too soon, skip withdrawal (but don't revert)
+        if (block.timestamp >= nextDistribution) {
```

Comment:
I think this if statement should be in the receive - so that sendFunds can be called at any time if required.

Reply:
this prevent dosing sendFunds by creating many withdrawal that need to be claimed on L1

Reply:
Ok, we should include a comment about this then

Reply:
Presumbly the same problem can be caused going the other way - create a lot of small retryable transactions that need to be redeemed. Perhaps in the specific case where the parent chain receiver is only receiving funds from the child chain then it's not a problem as they'll come in bulk. But it is a problem in the general case

Reply:
Also worth to note when the retryable expire, the callvalue would be refunded to the same destination address (tho this behavior may change). 

We may also consider hardcoding some minimum value used for gasLimit and gasPrice so that dosing is only possible when there is a L2 congestion over the predefined value (excess gas will always be refunded).